label_studio_converter/converter.py

```diff
@@ -424,13 +425,25 @@ def convert_to_coco(self, input_data, output_dir, output_image_dir=None, is_dir=
         for item_idx, item in enumerate(item_iterator):
             if not item['output']:
                 logger.warning('No annotations found for item #' + str(item_idx))
-                continue
+
+            image_id = len(images)
             image_path = item['input'][data_key]
             if not os.path.exists(image_path):
                 try:
                     image_path = download(image_path, output_image_dir, project_dir=self.project_dir,
                                           return_relative_path=True, upload_dir=self.upload_dir,
                                           download_resources=self.download_resources)
+                    
+                    #add image to list:
+                    with Image.open(os.path.join(output_dir, image_path)) as img:
```

Comment:
What should we do if images are on s3 bucket? Or if the converter can't have an access to them by security reasons? I think we can have this approach, but it's better to cover check access to this image file and if there is no success then switch to the previous variant of code.

Reply:
Good point, I did not consider this use case.

Won't the download function raise an exception here?:
https://github.com/heartexlabs/label-studio-converter/blob/39d308d31e8a9bd77ef5ef4005a099918bd22ab8/label_studio_converter/utils.py#L130

If I understand correctly then this image including annotations will be skipped, so if you have all your images on s3 the coco export will be empty?

For the future building the dataset client-side might be the best solution, but I can imagine that's a big thing to implement..

Reply:
@lfeldb 

> Won't the download function raise an exception here?
Looks like yes, I'd skip this raise and continue exporting.. 

> If I understand correctly then this image including annotations will be skipped, so if you have all your images on s3 the coco export will be empty?
yes, it will be empty.

Reply:
Agreed.

Could we not just use the cloud storage credentials server-side to get access to the images? That would be a nicer solution than building the dataset client-side imho.